include/posix/pthread.h

```diff
@@ -157,42 +159,61 @@ static inline int pthread_condattr_destroy(pthread_condattr_t *att)
 		.sem = &name##_psem,		\
 	}
 
-/**
- * @brief POSIX threading compatibility API
+/*
+ *  Mutex attributes - type
+ *
+ *  PTHREAD_MUTEX_NORMAL: Owner of mutex cannot relock it. Attempting
+ *      to relock will cause deadlock.
+ *  PTHREAD_MUTEX_RECURSIVE: Owner can relock the mutex.
+ *  PTHREAD_MUTEX_ERRORCHECK: If owner attempts to relock the mutex, an
+ *      error is returned.
  *
- * See IEEE 1003.1
  */
-static inline int pthread_mutex_init(pthread_mutex_t *m,
-				     const pthread_mutexattr_t *att)
-{
-	ARG_UNUSED(att);
+#define PTHREAD_MUTEX_NORMAL        0
+#define PTHREAD_MUTEX_RECURSIVE     1
+#define PTHREAD_MUTEX_ERRORCHECK    2
+#define PTHREAD_MUTEX_DEFAULT       PTHREAD_MUTEX_NORMAL
 
-	k_sem_init(m->sem, 1, 1);
+#define MUTEX_MAX_REC_LOCK 32767
 
-	return 0;
-}
+/* This timeout is not standard value. Set for convenience. */
+#define MEM_TIMEOUT 100
```

Comment:
This is an implementation detail and shouldn't be in pthread.h.  In fact, just use K_FOREVER where the memory is allocated; the POSIX APIs make no timing guarantees when allocating a mutex, so the user knows that the thread might block forever.

Reply:
Done